sylt/src/formatter.rs

```diff
@@ -0,0 +1,504 @@
+                write_statement(dest, indent + 1, s)?;
+                write!(dest, "\n")?;
+            }
+
+            write_indents(dest, indent)?;
+            write!(dest, "}}")?;
+        }
+        StatementKind::Break => write!(dest, "break")?,
+        StatementKind::Continue => write!(dest, "continue")?,
+        StatementKind::Definition {
+            ident,
+            kind,
+            ty,
+            value,
+        } => {
+            write_identifier(dest, ident)?;
+            if matches!(ty.kind, TypeKind::Implied) {
+                write!(
+                    dest,
+                    "{}",
+                    match kind {
+                        VarKind::Const => " :: ",
+                        VarKind::Mutable => " := ",
+                        VarKind::ForceConst => unreachable!("can't force an implied type"),
+                        VarKind::ForceMutable => unreachable!("can't force an implied type"),
+                    }
+                )?;
+            } else {
+                write!(dest, ": ")?;
+                match kind {
+                    VarKind::ForceConst | VarKind::ForceMutable => write!(dest, "!")?,
+                    VarKind::Const | VarKind::Mutable => (),
+                }
+                write_type(dest, indent, ty)?;
+                match kind {
+                    VarKind::Const | VarKind::ForceConst => write!(dest, " : ")?,
+                    VarKind::Mutable | VarKind::ForceMutable => write!(dest, " = ")?,
+                }
+            }
+            write_expression(dest, indent, value)?;
+        }
+        StatementKind::EmptyStatement => (),
+        StatementKind::If {
+            condition,
+            pass,
+            fail,
+        } => {
+            if matches!(fail.kind, StatementKind::EmptyStatement) {
+                for comment in &fail.comments {
+                    write!(dest, "// {}\n", comment)?;
+                    write_indents(dest, indent)?;
+                }
+            }
+
+            write!(dest, "if ")?;
+            write_expression(dest, indent, condition)?;
+            write!(dest, " ")?;
+            write_statement(dest, indent, pass)?;
+            if !matches!(fail.kind, StatementKind::EmptyStatement) {
+                write!(dest, " else ")?;
+                write_statement(dest, indent, fail)?;
+            } else {
+
+            }
+        }
+        StatementKind::IsCheck { lhs, rhs } => {
+            write_type(dest, indent, lhs)?;
+            write!(dest, " is ")?;
+            write_type(dest, indent, rhs)?;
+        }
+        StatementKind::Loop { condition, body } => {
+            write!(dest, "loop ")?;
+            write_expression(dest, indent, condition)?;
+            write!(dest, " ")?;
+            write_statement(dest, indent, body)?;
+        }
+        StatementKind::Ret { value } => {
+            write!(dest, "ret ")?;
+            write_expression(dest, indent, value)?;
+        }
```

Comment:
This won't write implicit returns - add a todo?

Reply:
I want to collect all todos in the same place, not spread out as comments.